metric/system/cgroup/util.go

```diff
@@ -229,7 +229,7 @@ func SubsystemMountpoints(rootfs resolve.Resolver, subsystems map[string]struct{
 
 // ProcessCgroupPaths returns the cgroups to which a process belongs and the
 // pathname of the cgroup relative to the mountpoint of the subsystem.
-func (r Reader) ProcessCgroupPaths(pid int) (PathList, error) {
+func (r Reader) ProcessCgroupPaths(version CgroupsVersion, pid int) (PathList, error) {
```

Comment:
A little ambivalent about how we're doing this. Right now, this function is basically version agnostic, and will return a list of both V1 and V2 cgroups, which are part of `PathList`. This might not matter for `GetV1StatsForProcess` and `GetV2StatsForProcess`, but `ProcessCgroupPaths` is called by the `add_docker_metadata` and `add_process_metadata` in beats, and I'm a little worried about us somehow changing the behavior of the processors under hybrid systems if we revert to either _only_ V1 or V2

My suggestion is that we create two functions (they can probably just be wrappers for the same private function or something) that can either version-agnostic, and return the full `PathList`, or that are version-specific, and just return the a map of the given version paths.

Reply:
Yeah, I wasn't sure if `ProcessCgroupPaths` was part of the public API for the module or not. As it's called  from beats, I'll rework the PR to not change the signature.

Reply:
I changed the approach in https://github.com/elastic/elastic-agent-system-metrics/pull/120/commits/ce87b3c8acb60c5f7d6a774f539a41b800804d0c to contain the changes in `ProcessCgroupPaths`. I had to call `CgroupsVersion` to know if we were in cgroup V1 or V2 mode. I could have refactored  `ProcessCgroupPaths` instead and collected all the cgroup controller before resolving paths, but I didn't want to copy the logic from `CgroupsVersion`